include/networkit/io/NetworkitBinaryGraph.hpp

```diff
@@ -28,5 +42,72 @@ static constexpr uint64_t DIR_MASK = 0x1; // bit 0
 static constexpr uint64_t WGHT_MASK = 0xE; //bit 1-3
 static constexpr uint64_t WGHT_SHIFT = 0x1;
 
+/// Serializes value into a buffer and returns the number of bytes written.
+inline size_t varIntEncode(uint64_t value, uint8_t *buffer) noexcept {
+	if (!value) {
+		buffer[0] = 1;
+		return 1;
+	}
+
+	const auto bits = 64 - tlx::clz(value);
+	auto dataBytes = ((bits - 1) / 7) - (bits == 64);
```

Comment:
I think this is quite hard to read. The problem is that the <= 56 bit cases and the > 56 bit case (first byte is zero) are fundamentally different: when we go from an encoded size of 8 to an encoded size of 9, we suddenly gain 8 bits of information instead of 7 bits (because the first byte is already fully consumed at that point; we go from 0 (first byte which is already `0b10000000`) + 7x8 = 56 bits to 0 (first byte which is now 0) + 8x8 = 64 bits).

Why not branch on `if(bits <= 56)` first? This would allow us to calculate `dataBytes = (bits - 1) / 7` in the then-case and to just set `dataBytes = 8` in the else-case. This was presumably the intention of the old code, even though the if-condition was indeed broken.